tooling/internal/cmd/configargs.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/knadh/koanf/maps"
+	"github.com/knadh/koanf/providers/structs"
+	"github.com/knadh/koanf/v2"
+	"github.com/microsoft/yardl/tooling/pkg/packaging"
+	"github.com/rs/zerolog/log"
+)
+
+var k = koanf.New(".")
```

Comment:
Slightly different implementation here: https://github.com/microsoft/tyger/blob/a838e141df846e373ef073d288aa731343db7e9a/cli/internal/cmd/install/installcommon.go#L43. Not suggesting it's better. Might want to check that we error out if overrides are missplelled etc.

Reply:
Nice, I will mimic this. I chose to ignore keys that are not valid `PackageInfo` fields (and log it, which can be seen in `--verbose` mode).

Do you mean, for example, to compute string edit distance and ask the user "Did you mean to type X instead of Y"?

Or just throw an error if the `-c key=value` override doesn't override anything?

Reply:
I would just raise an error.